ecs-cli/integ/cmd_up/up_test.go

```diff
@@ -0,0 +1,281 @@
+			Cluster:            aws.String(clusterName),
+			ContainerInstances: cluster.ContainerInstanceArns,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "unexpected error while describing container instances")
+		}
+		return instances.ContainerInstances, nil
+	}
+	return nil, errors.New(fmt.Sprintf("timeout after %d seconds while retrieving container instances from cluster %s", timeoutForWaitingOnActiveInstancesInS, clusterName))
+}
+
+// areInstancesActive returns true if all the instances have the status ACTIVE, false otherwise.
+func areInstancesActive(t *testing.T, instances []*ecs.ContainerInstance) bool {
+	for _, instance := range instances {
+		if !assert.Equal(t, *instance.Status, ecs.ContainerInstanceStatusActive) {
+			return false
+		}
+	}
+	return true
+}
+
+// areInstancesWithAttributes returns true if all the instances have the required attributes, false otherwise.
+func areInstancesWithAttributes(t *testing.T, instances []*ecs.ContainerInstance, attributes []*ecs.Attribute) bool {
+	for _, instance := range instances {
+		instanceAttrs := make(map[string]*ecs.Attribute)
+		for _, attr := range instance.Attributes {
+			instanceAttrs[*attr.Name] = attr
+		}
+
+		for _, wantedAttr := range attributes {
+			actualAttr, hasKey := instanceAttrs[*wantedAttr.Name]
+			if !hasKey {
+				t.Logf("instance %s is missing attribute name %s", *instance.Ec2InstanceId, *wantedAttr.Name)
+				return false
+			}
+			if !assert.Equal(t, wantedAttr, actualAttr) {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// areInstancesWithResourceNames returns true if all the instances have the required resource names, false otherwise.
+func areInstancesWithResourceNames(t *testing.T, instances []*ecs.ContainerInstance, resourceNames []string) bool {
+	for _, instance := range instances {
+		instanceResources := make(map[string]bool)
+		for _, resource := range instance.RegisteredResources {
+			instanceResources[*resource.Name] = true
+		}
+
+		for _, wantedName := range resourceNames {
+			if _, hasKey := instanceResources[wantedName]; !hasKey {
+				t.Logf("instance %s is missing resource name %s", *instance.Ec2InstanceId, wantedName)
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// after best-effort deletes any resources created by the upTest.
+func after(cfnClient *cloudformation.CloudFormation, ecsClient *ecs.ECS, clusterName string) {
+	deleteStack(cfnClient, clusterName)
+	deleteCluster(ecsClient, clusterName)
+}
+
+func deleteStack(client *cloudformation.CloudFormation, clusterName string) {
+	client.DeleteStack(&cloudformation.DeleteStackInput{
+		StackName: aws.String(stackName(clusterName)),
+	})
+}
+
+func deleteCluster(client *ecs.ECS, clusterName string) {
+	client.DeleteCluster(&ecs.DeleteClusterInput{
+		Cluster: aws.String(clusterName),
+	})
+}
+
+func stackName(clusterName string) string {
+	const ecsCLIStackNamePrefix = "amazon-ecs-cli-setup-"
```

Comment:
We already define this is as const in [/modules/utils](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/utils/utils.go#L29) - not sure how you feel about importing modules from the cli into this package, but might be handy to use this in case it ever changes.

Reply:
For visibility, we discussed this offline:
> I think the integ tests (unlike unit tests) should try to be as little reliant as possible to the cli code since we are trying to test that. In case we introduce a weird typo in the prefix in the cli, these tests should catch that.